tests/hash_table_test.c

```diff
@@ -1258,3 +1258,39 @@ static int s_test_hash_table_byte_cursor_create_find_fn(struct aws_allocator *al
 
     return 0;
 }
+
+AWS_TEST_CASE(test_hash_combine, s_test_hash_combine_fn)
+static int s_test_hash_combine_fn(struct aws_allocator *allocator, void *ctx) {
+    uint64_t a, b, c;
+
+    /* We're assuming that the underlying hashing function works,
+     * just making sure we hooked it up right for 2 64bit values */
+
+    a = 0x123456789abcdef;
+    b = 0xfedcba987654321;
+    c = aws_hash_combine(a, b);
+
+    /* Sanity check */
+    ASSERT_TRUE(c != a);
+    ASSERT_TRUE(c != b);
+
+    /* Same inputs gets same results, right? */
+    ASSERT_UINT_EQUALS(c, aws_hash_combine(a, b));
+
+    /* Using all bytes, right? */
+    uint8_t *c_bytes = (uint8_t *)&c;
+    for (size_t i = 0; i < sizeof(c); ++i) {
+        ASSERT_TRUE(c_bytes[i] != 0);
+    }
+
+    /* Check even low-entropy numbers result in something with high entropy */
+    c = aws_hash_combine(0, 1);
+    for (size_t i = 0; i < sizeof(c); ++i) {
+        ASSERT_TRUE(c_bytes[i] != 0);
```

Comment:
Can we do better than that? If we want entropy, we could actually measure entropy.

Reply:
Yes we could do better, but this is not the test were we would do it.

We use `lookup3.inl`, an existing collection of functions for hashing. We've assumed so far that these do what they say. I just wanted to make sure I didn't do something dumb where the result never had its high or low bytes set.

Reply:
I just took this part of the test out. It was a naive test the quality of the underlying hash function, and this isn't the place for it